### PR TITLE
Update docs/trusting-cert-per-language for Mac troubleshooting

### LINKED ISSUE #2090: [Test-Proxy] Current default dev certificate will not work on `mac`
Shown in [this commit](https://github.com/scbedd/azure-sdk-for-python/blob/873624df6ea5a3fdff691330de45f969f456c1a7/eng/pipelines/test-with-proxy.yml#L70).

We exported our `pfx` using `pkcs12` format, while the mac keychain explicitly requires `x509`. We need to either re-roll a cross-compatible one, or simply split it out and use it.

### PR DESCRIPTION
**Pull request by jeesang7**
Fix #2090 

On Mac, add keychain description in case of permission errors when executing `dotnet dev-certs` related.
Any feedback is welcome. Thanks.

### CODE REVIEW

**Comment by scbedd** [comment]
@jeesang7 thank you so much for the PR! I'll dig up a mac machine, double check your guidance (literally just due diligence) and then merge this. 

Seriously, always appreciate documentation PRs.

**Comment by scbedd** [comment]
@jeesang7 I just now was able to hunt down a mac machine. I'll review this ASAP.

**Review by ianychoi — COMMENTED** [review]
Thank you this PR @jeesang7 during our open source contribution journey!

@scbedd for your information, me and @yeongseon are encouraging open source contributions to https://github.com/Azure/azure-sdk-for-python repo through mentoring and were explaining how to install and test SDK source repos. Any helpful comments on this, or your call for open source contributions on SDK tool chain or Python related parts would be great :)

File: tools/test-proxy/documentation/trusting-cert-per-language.md [hunk 1]
```diff
@@ -23,6 +23,24 @@ dotnet dev-certs https --trust
...
+Cleaning HTTPS development certificates from the machine. This operation might require elevated privileges. If that is the case, a prompt for credentials will be displayed.
+HTTPS development certificates successfully removed from the machine.
+The provided certificate file 'eng/common/testproxy/dotnet-devcert.pfx' is not a valid PFX file or the password is incorrect.
+```
+
+In this case, you can manually set it in 'Keychain Access' to work around the problem.
+1. Click the `dotnet-devcert.pfx` file in 'Finder' to register the keychain directly. Enter the password as “password”
+2. You can check the newly created `localhost` name in the keychain access “system” item
+3. Double-click `localhost` and change Trust to "Always Trust"
+4. Run `$ dotnet dev-certs https —trust` in the terminal, and you can see that the `localhost` checked above has changed from ![x](https://img1.daumcdn.net/thumb/R1280x0/?scode=mtistory2&fname=https%3A%2F%2Fblog.kakaocdn.net%2Fdn%2FMdyzC%2FbtrIc9Azung%2F83TrRVVDYVaf85gbiAL2VK%2Fimg.png) to ![+](https://img1.daumcdn.net/thumb/R1280x0/?scode=mtistory2&fname=https%3A%2F%2Fblog.kakaocdn.net%2Fdn%2FbdsBJ8%2FbtrIeMklVrv%2FeHV4xcBVvT476nxofIuaHK%2Fimg.png)
```
[the remark this answers is not in the document]
Comment (jeesang7) [reply tools/test-proxy/documentation/trusting-cert-per-language.md]: That's a good idea. Thank you.

**Review by scbedd — COMMENTED** [review]
Ok. I repro-ed all of this on my mac machine.

For all your additional detail, excellent! I was able to follow along with 0 issues.

However, I tried to _use_ the certificate after install and trust according to your method, and unfortunately the `test-proxy` doesn't load up the certificate when booting.

In my experience, it only ever binds the http port of 5000, so the effort doesn't get us any closer to using the cert on a mac.

It is _extremely_ possible that I'm missing something though. Have you tried running the `test-proxy` your machine? Was it able to bind https to port 5001?

**Comment by jeesang7** [comment]
First, there is a typo(52aac8d).  Can you check with a double dash for the `trust` argument?
```
dotnet dev-certs https --trust
```

And, the test-proxy I checked was a way to run pytest [based on docker](
52aac8d). The .NET based test-proxy is still being reviewed on my macOS.
I'm looking for the usage of `test-proxy --storage-location <location>` command.

**Comment by ponopono0322** [comment]
Have you ever seen like this(5000 to 0) error?
`Ports are not available: exposing port TCP 0.0.0.0:5000 -> 0.0.0.0:0: listen tcp 0.0.0.0:5000: bind: address already in use`(this error come from docker)

In this case, you should check port is already using.
Open terminal, and copy and paste command

```bash
sudo lsof -i :5000
```
to check port using.

If you seen NAME `*:commplex-main (LISTEN)`, this process come from mac system.
Following next step to deactivate `Airplay Recevier`.

Open `System Preference` > `Sharing` > Uncheck `Airplay Recevier`
and check again port(using `lsof`).

Also, other process were opened, kill process:
```bash
sudo kill PID
```
(PID is number)

In this way, i could run pytest via docker.

This scenario was performed on the m1 mac Monterey.

**Review by scbedd — APPROVED** [review]

**Comment by scbedd** [comment]
Thank you for the additional detail and detailed feedback on possible errors! It helps a ton.

**Pull request merged**